configure.ac

```diff
@@ -486,6 +486,25 @@ AC_ARG_WITH([cuda],
 			    and runtime libraries are installed.])],
 	    [], [])
 
+have_libcuda=0
+AS_IF([test x"$with_cuda" != x"no"],
+	    [FI_CHECK_PACKAGE([cuda],
+			      [cuda_runtime.h],
+			      [cudart],
+			      [cudaMemcpy],
+			      [-lcuda],
+			      [$with_cuda],
+			      [],
+			      [have_libcuda=1],
+			      [],
+			      [])],
+	    [])
+
+AS_IF([test "$with_cuda" = "yes" && test "$have_libcuda" = "0" ],
+	[AC_MSG_ERROR([CUDA support requested but CUDA runtime not available.])],
+	[])
```

Comment:
FWIW, you don't need these empty blocks.  You can keep them if you like them for style, but they aren't actually necessary (in AS_IF or FI_CHECK_PACKAGE).

Reply:
Applied the "do what the rest of the code does" principle for style here.